### ISSUE TITLE
Redirecting to URL with explicit port causes request to fail

### ISSUE THREAD

**Issue by lukeredpath** [pokeb/asi-http-request#39]
In my test example, I'm requesting a URL, say:

http://example.com/foo

Which redirects to:

http://example.com:80/foo

This fails because an exception is thrown deep in the guts of ASIHTTPRequest. I tracked this down to line 2904 of ASIHTTPRequest.m in the method `findSessionAuthenticationCredentials`, where it tries to compare the port of the new URL (80) to the port in the original URL (nil). The exception is thrown because it is calling `NSNumber isEqualToNumber:` with a nil argument.

So, this line:

```
        if ([[theURL host] isEqualToString:[requestURL host]] && ([theURL port] == [requestURL port] || [[theURL port] isEqualToNumber:[requestURL port]]) && [[theURL scheme] isEqualToString:[requestURL scheme]]) {
```

Should probably be something like (ugly, but works):

```
        if ([[theURL host] isEqualToString:[requestURL host]] && ([theURL port] == [requestURL port] || ([requestURL port] && [[theURL port] isEqualToNumber:[requestURL port]])) && [[theURL scheme] isEqualToString:[requestURL scheme]]) {
```

**Comment by pokeb** [comment]
Fix a problem that could occur in searching for credentials when mixing urls containing a port with ones that don't -  Closed by 813811a63b9d5ad37b3baebd03abd9951ae6bdb5
Many thanks to Luke Redpath for tracking this down!

**Comment by pokeb** [comment]
Hi Luke

Thanks very much for letting me know, this should be fixed now.

Best

Ben
